Source/WebCore/rendering/svg/SVGResourcesCycleSolver.h

```diff
@@ -38,12 +38,13 @@ class SVGResourcesCycleSolver {
     void resolveCycles();
 
 private:
-    bool resourceContainsCycles(RenderElement&) const;
+    bool resourceContainsCycles(RenderSVGResourceContainer&);
     void breakCycle(RenderSVGResourceContainer&);
 
     RenderElement& m_renderer;
     SVGResources& m_resources;
-    WeakHashSet<RenderSVGResourceContainer> m_allResources; 
+    WeakHashSet<RenderSVGResourceContainer> m_activeResources;
+    WeakHashSet<RenderSVGResourceContainer> m_acyclicResources;
```

Comment:
I would suggest making SVGResourcesCycleSolver no constructible class and making all its methods be static functions. So the caller can say:

```
    SVGResourcesCycleSolver::resolveCycles(renderer, resources);
```
instead of 

```
    SVGResourcesCycleSolver solver(renderer, resources);
    solver.resolveCycles();
```

All these members can be just arguments passed from one function to another.

Reply:
Okay, done.